src/examples/3d_fluid_sim/fluid_sim.cpp

```diff
@@ -0,0 +1,198 @@
+#include "fluid_sim.hpp"
+
+#define simIterations 250
+#define DECLARE_VARIABLES \
+    wash::add_variable("deltaTime",  (1.0/60.0) / 3.0 * 0.9 ); \
+    wash::add_variable("gravity", -10.0); \
+    wash::add_variable("collisionDamping", 0.95); \
+    wash::add_variable("smoothingRadius", 0.2); \
+    wash::add_variable("targetDensity", 630.0); \
+    wash::add_variable("pressureMultiplier", 288.0); \
+    wash::add_variable("nearPressureMultiplier", 2.25); \
+    wash::add_variable("viscosityStrength", 0.08);
+#define numParticles 42875
```

Comment:
I would declare these as e.g. `constexpr uint64 numParticles = 42875;`

Reply:
This was a fairly non-trivial transformation involving updating Vec constructors to be constexpr, and adding a new const Vec multiplication operator override (probably have to eventually implement more of those) but done. This should probably help the performance as well.